logging/new_logger

```diff
@@ -0,0 +1,28 @@
+# New source code logger
+
+## Current logger
+
+* fc::logger
+  ** Console logging both stderr & stdout
+  ** GELF logging
+  ** Deep-mind logging to a file
+  ** Works with fc::variant for logging any FC_REFLECT type
+  
+Could enhance fc::logger to add file support, file rotation, async logging, filtering, etc. OR switch to a third-party open source logger.
+
+## New logger
+
+Evaluate existing open source log libraries. `spdlog` (https://github.com/gabime/spdlog) has been used by core team in the past on other projects.
+
+## Integration
+
+`spdlog` uses `fmt` library which is part of C++20 or is available as a separate third-party library (https://github.com/fmtlib/fmt).
```

Comment:
It is part of c++20 but sadly is still not present in the latest versions of libstdc++ nor libc++, so realistically we'll need to use the third party library for the next 4+ years.

Reply:
> so realistically we'll need to use the third party library for the next 4+ years.

Seems to me that it would be better than the alternative (of not using `std::format` or `fmtlib`)